lib/internal/modules/esm/resolve.js

```diff
@@ -90,6 +90,18 @@ function emitFolderMapDeprecation(match, pjsonUrl, isExports, base) {
   );
 }
 
+function emitLegacyIndexTypeModuleDeprecation(packageJSONUrl, base) {
+  if (!pendingDeprecation)
+    return;
+  const pkgPath = fileURLToPath(new URL('.', packageJSONUrl));
+  const basePath = fileURLToPath(base);
+  process.emitWarning(
+    `Package ${pkgPath} has a "type": "module" field in its package.json, but` +
+    ` does not define an "exports" field, imported from ${basePath}.\nLegacy ` +
+    'main resolution lookups are deprecated for packages with "type": "module".'
```

Comment:
```suggestion
    ` does not define an "exports" field, imported from ${basePath}.\n` +
    'Packages with "type": "module" must define an "exports" field; ' +
    'defining only "main" is deprecated.'
```

Reply:
Let's come back to this after further discussion, while the semantics are still up in the air.